hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -126,10 +134,16 @@
   private long defaultBandwidth;
   private long currentBandwidth;
   private WALFileLengthProvider walFileLengthProvider;
+
   @VisibleForTesting
-  protected final ConcurrentHashMap<String, ReplicationSourceShipper> workerThreads =
+  protected final Map<String, ReplicationSourceShipper> workerThreads =
       new ConcurrentHashMap<>();
 
+  private final ListeningExecutorService executorService =
```

Comment:
This could use some comments (what is in this executor service?).. While you are here, mind adding some javadocs for the workerThreads too?